### ISSUE TITLE
Unable to Transform publishing page with OOTB web parts whereas those web parts are upgraded when the page was not publishing type

### ISSUE THREAD

**Issue by RajiVenkat242** [SharePoint/sp-dev-modernization#239]
Thank you for reporting an issue or suggesting an enhancement. We appreciate your feedback - to help the team to understand your needs, please complete the below template to ensure we have the necessary details to assist you. Thanks!

#### Category

[ x] Bug
[ ] Enhancement

#### Problem Area

[ ] Page Transformation: Error during the setup/use of the Page Transformation UI solution (did you check our [troubleshooting guide](https://github.com/SharePoint/sp-dev-modernization/blob/dev/Solutions/PageTransformationUI/docs/troubleshootingguide.md)?)
[x ] Page Transformation: Error during the use of page transformation from PnP PowerShell
[ ] Page Transformation: Error during the use of page transformation from .Net
[ ] Page Transformation: Page is not looking correct after transformation
[ ] Modernization Scanner: something went wrong...

---------------------

Hi, 

I am trying to upgrade a publishing page using PnP PowerShell command as shown below. Text web part is upgraded. But Calendar web part is not transformed. 

ConvertTo-PnPClientSidePage -Identity "WebPart_TestPage_R2.aspx" -AddPageAcceptBanner -PublishingPage -TargetWebUrl $communicationsiteurl -PublishingTargetPageName $item.Name -LogType File -LogFolder "D:\Scripts\log" -LogVerbose

It is failed when trying to map the calendar list web part to Events web part in a publishing page. 
This calendar web part is open in the classic page.

Please find the transformation log. 

# Modernisation Summary Report

Date  |  Duration  |  Source Page  |  Target Page Url  |  Status
-------------  |  -------------  |  -------------  |  -------------  |  -------------
8/20/2019 2:06:00 AM  |  00:00:02  |  [/pages/webpart_testpage_r2.aspx](teams/it/pages/webpart_testpage_r2.aspx)  |  []()  |  A issue prevented successful transformation
## Errors during transformation

Date  |  Source Page  |  Operation  |  Message
-------------  |  -------------  |  -------------  |  -------------
8/20/2019 2:06:02 AM  |  /pages/webpart_testpage_r2.aspx  |   Web Part Mapping  |  An error occurred processing functions - Exception has been thrown by the target of an invocation.    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at SharePointPnP.Modernization.Framework.Functions.BaseFunctionProcessor.ExecuteMethod(Object functionClassInstance, FunctionDefinition functionDefinition, MethodInfo methodInfo)
   at SharePointPnP.Modernization.Framework.Functions.FunctionProcessor.ExecutePropertyFunctions(String[] functionsToProcess, WebPart webPartData, WebPartEntity webPart, Property property)
   at SharePointPnP.Modernization.Framework.Functions.FunctionProcessor.Process(WebPart& webPartData, WebPartEntity webPart)
   at SharePointPnP.Modernization.Framework.Transform.ContentTransformator.Transform(List`1 webParts)
   at SharePointPnP.Modernization.Framework.Publishing.PublishingPageTransformator.Transform(PublishingPageTransformationInformation publishingPageTransformationInformation)
8/20/2019 2:06:02 AM  |  /pages/webpart_testpage_r2.aspx  |   Summary  |  A critical error occurred - transformation did not complete    at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at SharePointPnP.Modernization.Framework.Functions.BaseFunctionProcessor.ExecuteMethod(Object functionClassInstance, FunctionDefinition functionDefinition, MethodInfo methodInfo)
   at SharePointPnP.Modernization.Framework.Functions.FunctionProcessor.ExecutePropertyFunctions(String[] functionsToProcess, WebPart webPartData, WebPartEntity webPart, Property property)
   at SharePointPnP.Modernization.Framework.Functions.FunctionProcessor.Process(WebPart& webPartData, WebPartEntity webPart)
   at SharePointPnP.Modernization.Framework.Transform.ContentTransformator.Transform(List`1 webParts)
   at SharePointPnP.Modernization.Framework.Publishing.PublishingPageTransformator.Transform(PublishingPageTransformationInformation publishingPageTransformationInformation)
### 8/20/2019 2:06:02 AM - [/pages/webpart_testpage_r2.aspx](/teams/it/pages/webpart_testpage_r2.aspx)

_Exception has been thrown by the target of an invocation.   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
   at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at SharePointPnP.Modernization.Framework.Functions.BaseFunctionProcessor.ExecuteMethod(Object functionClassInstance, FunctionDefinition functionDefinition, MethodInfo methodInfo)
   at SharePointPnP.Modernization.Framework.Functions.FunctionProcessor.ExecutePropertyFunctions(String[] functionsToProcess, WebPart webPartData, WebPartEntity webPart, Property property)
   at SharePointPnP.Modernization.Framework.Functions.FunctionProcessor.Process(WebPart& webPartData, WebPartEntity webPart)
   at SharePointPnP.Modernization.Framework.Transform.ContentTransformator.Transform(List`1 webParts)
   at SharePointPnP.Modernization.Framework.Publishing.PublishingPageTransformator.Transform(PublishingPageTransformationInformation publishingPageTransformationInformation)_ 

# Individual Page details
## Transformation Details: /pages/webpart_testpage_r2.aspx

- Report date: 8/20/2019 2:06:00 AM
- Transform duration: SharePointPnP.Modernization.Framework.Entities.TransformationLogAnalysis
- Transforming site: /teams/IT
- Cross-Site transfer mode to site: /sites/Comm9
- Transforming page: /teams/it/pages/webpart_testpage_r2.aspx
-  A critical error occurred - transformation did not complete

### Page Transformation Settings

Property  |  Setting
-------------  |  -------------
Overwrite   |   False
Target Page Name   |   WebPart_TestPage_R2.aspx
Keep Page Specific Permissions   |   True
Remove Empty Sections And Columns   |   True
Handle Wiki Images And Videos   |   True
Publish Created Page   |   True
Disable Page Comments   |   False
Skip Url Rewrite   |   False
Skip Telemetry   |   False
### Transformation Operation Details

Date  |  Operation  |  Actions Performed
-------------  |  -------------  |  ------------- 
8/20/2019 2:06:00 AM  |  _Input Validation_  |  _Validation checks complete_
8/20/2019 2:06:00 AM  |  _SharePoint Connection_  |  _Loading client context objects_
8/20/2019 2:06:00 AM  |  _SharePoint Connection_  |  _Loading target client context object_
8/20/2019 2:06:01 AM  |  _Page Creation_  |  _Detect if the page is living inside a folder_
8/20/2019 2:06:01 AM  |  _Page Creation_  |  _Just try to load the page in the fastest possible manner, we only want to see if the page exists or not_
8/20/2019 2:06:02 AM  |  Load  |  Page does not exist in current web
8/20/2019 2:06:02 AM  |  Page Creation  |  Checking Page Exists
8/20/2019 2:06:02 AM  |  Page Creation  |  Modern page created 
8/20/2019 2:06:02 AM  |  Article page handling  |  Transforming source page as Article page
8/20/2019 2:06:02 AM  |  Article page handling  |  Recognized source page as a Publishing Page. - Analyzing web parts and page layouts
8/20/2019 2:06:02 AM  |  Article page handling  |  Splitting images and videos from wiki text - as modern text web part does not support embedded images and videos
8/20/2019 2:06:02 AM  |  Set Page Title  |  Setting the modern page title: WebPart_TestPage
8/20/2019 2:06:02 AM  |  _Article page handling_  |  _Preparing content transformation_
8/20/2019 2:06:02 AM  |  Article page handling  |  Transforming content
8/20/2019 2:06:02 AM  |  Content Transform  |  Transforming web parts
8/20/2019 2:06:02 AM  |  Web Part Mapping  |  Web Part:'WikiText' of type 'SharePointPnP.Modernization.WikiTextPart' is being transformed
8/20/2019 2:06:02 AM  |  _Web Part Mapping_  |  _Processing selector functions_
8/20/2019 2:06:02 AM  |  _Web Part Mapping_  |  _Combining mapping data_
8/20/2019 2:06:02 AM  |  Adding Web Parts to Target Page  |  Added 'Client Side Text Web Part' to target page
8/20/2019 2:06:02 AM  |  Web Part Mapping  |  Web Part:'RCal' of type 'Microsoft.SharePoint.WebPartPages.ListViewWebPart' is being transformed
8/20/2019 2:06:02 AM  |  _Web Part Mapping_  |  _Processing selector functions_

**Comment by jansenbe** [comment]
Thx for reporting this @RajiVenkat242 , I'll look into this and provide feedback.

**Comment by jansenbe** [comment]
Hi @RajiVenkat242 ,

When you transform a publishing page holding a calendar web part we assume you've setup a calendar list with the same name in the target site collection. In your case this is not the case and the code failed on this. Other lists (XSLTListViewWebPart based) handled this situation and simply skipped the web part, but for ListViewWebPart based lists (like Calendar) this was not the case. I've updated the mapping to handle this correctly now.

So in summary for you the fix will to create the calendar list in the target site using the same name as was used in the source site.

**Comment by jansenbe** [comment]
Hi @RajiVenkat242 ,

Did above help? Can we close this issue?

**Comment by jansenbe** [comment]
Hey @RajiVenkat242 ,

Did the above feedback get you unblocked?

**Comment by jansenbe** [comment]
Since the fix to handle non existing calendar lists in the target site is checked in I'm closing this issue. Please create a new issue if you still face problems when using the September release which will be released soon.
